services/src/test/java/org/apache/druid/server/AsyncQueryForwardingServletTest.java

```diff
@@ -568,6 +595,18 @@ protected void doService(
 
     servlet.service(requestMock, null);
 
+    // NPE is expected since the listener's onComplete calls the parent class' onComplete which fails due to
```

Comment:
this part is not clear. can you elaborate on
- why servlet is in partial state? 
- NPE itself is an error. what other possible errors are anticipated? 
- where exactly does this NPE occur?

Reply:
1. It is in partial state since the `init` method is not called on the servlet. Initializing the servlet would require multiple things like `ServletConfig` and `HttpClientProvider`
2. So far I've seen NPE only due to uncreated objects in the servlet
3. Currently, it occurs while doing operations on the logger

Reply:
Sorry for the late reply. Changes LGTM. Though we should remove the assertion on NPE and just catch/ignore it. Since we are not really verifying the NPE itself. We are getting NPE because of the way, the test is written and not something that code does intentionally.

Reply:
makes sense, added a catch clause for NPE instead of an assertion.